### PR TITLE
Fix resolving common required fields in allOf schema

### PR DESCRIPTION
**Pull request by TharmiganK**
Fixes: https://github.com/ballerina-platform/ballerina-library/issues/5059

### CODE REVIEW

File: openapi-core/src/main/java/io/ballerina/openapi/core/generators/type/generators/AllOfRecordTypeGenerator.java [hunk 1]
```diff
@@ -205,6 +223,49 @@ private ImmutablePair<List<Node>, List<Schema<?>>> generateAllOfRecordFields(Lis
...
+        }
+        List<Schema<?>> allOfSchemas = schema.getAllOf();
+        if (Objects.nonNull(allOfSchemas)) {
+            for (Schema<?> allOfSchema : allOfSchemas) {
+                properties.putAll(getPropertiesFromRefSchema(allOfSchema));
+            }
+        }
+        return properties;
+    }
+
```
Comment (lnash94) [inline openapi-core/src/main/java/io/ballerina/openapi/core/generators/type/generators/AllOfRecordTypeGenerator.java:268]: Shall we add a test for this?

Comment (TharmiganK) [reply openapi-core/src/main/java/io/ballerina/openapi/core/generators/type/generators/AllOfRecordTypeGenerator.java:268]: Added via [91ce874](https://github.com/ballerina-platform/openapi-tools/pull/1712/commits/91ce8740930bbdbc2e3588e2569243e8207401f1)

File: openapi-core/src/main/java/io/ballerina/openapi/core/generators/type/generators/AllOfRecordTypeGenerator.java [hunk 2]
```diff
@@ -106,6 +110,10 @@ public TypeDescriptorNode generateTypeDescriptorNode() throws OASTypeGenExceptio
         // filtering as input. Has to use this assertion statement instead of `if` condition, because to avoid
         // unreachable else statement.
         List<Schema<?>> allOfSchemas = schema.getAllOf();
+        List<String> requiredFields = schema.getRequired();
+        if (Objects.isNull(requiredFields)) {
+            requiredFields = new ArrayList<>();
```
Comment (ayeshLK) [inline openapi-core/src/main/java/io/ballerina/openapi/core/generators/type/generators/AllOfRecordTypeGenerator.java:115]: What if we do this ?
```suggestion
            requiredFields = Collections.emptyList();
```

Comment (ayeshLK) [inline openapi-core/src/main/java/io/ballerina/openapi/core/generators/type/generators/AllOfRecordTypeGenerator.java:115]: Unless we are planning to modify the `requiredFields` variable we can do this.
```suggestion
            requiredFields = Collections.emptyList();
```

**Review by lnash94 — DISMISSED** [review]

**Review by NipunaRanasinghe — APPROVED** [review]

**Review by lnash94 — APPROVED** [review]

**Pull request merged**